rust/sbp/src/sbp_string.rs

```diff
@@ -37,6 +38,93 @@ where
     }
 }
 
+impl<T: AsRef<[u8]>> SbpString<T, Unterminated> {
+    /// Checked unterminated SbpString builder, verify last byte is non null
+    pub fn unterminated(data: T) -> Result<Self, UnterminatedError> {
+        match data.as_ref().last() {
+            Some(l) if l != &0 => Ok(Self::new(data)),
+            _ => Err(UnterminatedError),
+        }
+    }
+}
+
+impl<T: AsRef<[u8]>> SbpString<T, NullTerminated> {
+    pub fn null_terminated(data: T) -> Result<Self, NullTerminatedError> {
+        match data.as_ref().last() {
+            Some(l) if l == &0 => Ok(Self::new(data)),
+            _ => Err(NullTerminatedError),
+        }
+    }
+}
+
+/// Helper function to alternate null bytes
+/// Used in [`SbpString::from_parts`]
+///
+/// Ex. [a, b, c] => [a, 0, b, 0, c, 0]
+///
+/// Returns null terminated vector
+fn alt_null<R, I>(parts: I) -> Vec<u8>
+where
+    R: AsRef<[u8]>,
+    I: IntoIterator<Item = R>,
+{
+    let mut data = vec![];
```

Comment:
I'd be tempted to write some iterator nonsense here but this seems really readable actually 👌

Reply:
```rust
    parts
        .into_iter()
        .flat_map(|r| {
            r.as_ref()
                .iter()
                .flat_map(|i| vec![*i, 0])
                .collect::<Vec<_>>()
        })
        .collect()
```

there "fixed" 😆

Reply:
would have loved `Iterator::intersperse` but it is nightly :(